astropy/io/fits/hdu/hdulist.py

```diff
@@ -1065,6 +1074,29 @@ def _readfrom(cls, fileobj=None, data=None, mode=None, memmap=None,
             # fromstring case; the data type of ``data`` will be checked in the
             # _BaseHDU.fromstring call.
 
+        if hdulist._file and hdulist._file.readonly:
+            pos = hdulist._file.tell()
+            raise_error = False
+
+            try:
+                simple = hdulist._file.read(30).decode('ascii')
+            except UnicodeDecodeError:
+                raise_error = True
+            else:
+                if simple != 'SIMPLE  =                    T':
+                    if simple == 'SIMPLE  =                    F':
```

Comment:
Instead of calling `decode('ascii')` above could it not just compare directly to `FITS_SIGNATURE[:-1]` plus ensure `simple[-1] in (b'T', b'F')`?

Reply:
Yeah indeed, I think I went that way because of one of the unit test using some unicode characters, but I agree it can be simplified!